File: 프로그래머스/Python/체육복.py
def solution(n, lost, reserve):
    answer = n - len(lost)

    lost.sort()
    reserve.sort()

    for i in range(len(lost)):
        if lost[i] in reserve:
            reserve.remove(lost[i])
            lost[i] = -1
            answer += 1

    for i in range(len(lost)):

        if lost[i] in reserve:
            lost.remove(lost[i])
            reserve.remove(lost[i])
            answer += 1
        if lost[i] - 1 in reserve:
            answer += 1
            reserve.remove(lost[i] - 1)
            continue
        if lost[i] + 1 in reserve:
            answer += 1
            reserve.remove(lost[i] + 1)
            continue

    return answer

File: 프로그래머스/lv1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        Set<Integer> lostSet = new HashSet<>();
        Set<Integer> reserveSet = new HashSet<>();

        // 여벌 가져온 사람
        for (int i : reserve) {
            reserveSet.add(i);
        }

        for (int i : lost) {
            // 여벌 가져온 사람이 잃어버린 경우
            if (reserveSet.contains(i)) {
                reserveSet.remove(i);
            } else {
                lostSet.add(i);
            }
        }

        // 잃어버린 사람에게 빌려줄 수 있는 경우
        for (Integer i : reserveSet) {
            if (lostSet.contains(i - 1)) {
                lostSet.remove(i - 1);
            } else if (lostSet.contains(i + 1)) {
                lostSet.remove(i + 1);
            }
        }

        return n - lostSet.size();
    }
}